types/mempool/sender_nonce.go

```diff
@@ -0,0 +1,199 @@
+package mempool
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/signing"
+	huandu "github.com/huandu/skiplist"
+)
+
+var (
+	_ Mempool  = (*senderNonceMempool)(nil)
+	_ Iterator = (*nonceMempoolIterator)(nil)
+)
+
+type senderTxs struct {
+	txQueue *huandu.SkipList
+	head    *huandu.Element
+}
+
+func newSenderTxs() senderTxs {
+	return senderTxs{
+		head:    nil,
+		txQueue: huandu.New(huandu.LessThanFunc(txKeyLessNonce)),
```

Comment:
We don't need a comparator function if we're just comparing nonces. 
```suggestion
		txQueue: huandu.New(huandu.Uint64),
```